components/SearchBar.tsx

```diff
@@ -7,35 +7,16 @@ import {
 import { useState, KeyboardEvent } from 'react'
 import { SearchIcon } from './Icons'
 
-interface SearchBarProps<T> extends InputGroupProps {
-  baseData: T[]
-  setData: (data: T[]) => void
-  searchFields: string[]
+interface SearchBarProps extends InputGroupProps {
+  handleSearch: (query: string) => void
 }
 
-export const SearchBar = <T,>({
-  baseData,
-  setData,
-  searchFields,
+export const SearchBar = ({
+  handleSearch,
+  placeholder,
   ...props
-}: SearchBarProps<T>): JSX.Element => {
+}: SearchBarProps): JSX.Element => {
   const [query, setQuery] = useState('')
-  const handleSearch = (query: string): void => {
-    if (query === '') {
-      setData(baseData)
-      return
-    }
-
-    setData(
-      baseData.filter((datum) =>
-        searchFields.some((field) =>
-          (datum[field as keyof T] as unknown as string)
-            ?.toLowerCase()
-            .includes(query.toLowerCase())
-        )
-      )
-    )
-  }
```

Comment:
Doing the filtering on the base data here in the `SearchBar` would override any filtering done through the taxonomies dropdown so I refactored searching and filtering by taxonomies into a `useFilters()` hook that stacks those functionalities.